test/task.cc

```diff
@@ -299,31 +390,27 @@ TEST(Task, StartFuture) {
 }
 
 TEST(Task, FailContinuation) {
-  auto e = []() -> Task::Of<int> {
+  auto e = []() -> Task::Of<int>::Raises<std::runtime_error> {
     return [x = 42]() {
       return Just(x);
     };
   };
 
-  static_assert(
-      eventuals::tuple_types_unordered_equals_v<
-          decltype(e())::ErrorsFrom<void, std::tuple<>>,
-          std::tuple<>>);
-
   std::optional<Task::Of<int>::Raises<std::runtime_error>> task;
 
   task.emplace(e());
 
-  std::exception_ptr result;
+  std::optional<std::runtime_error> result;
 
   task->Fail(
       GenerateTestTaskName(),
       std::runtime_error("error"),
       [](int) {
         FAIL() << "test should not have succeeded";
       },
-      [&](std::exception_ptr exception) {
-        result = std::move(exception);
+      [&](std::variant<Stopped, std::runtime_error>&& exception) {
```

Comment:
This looks wrong, we should only be passing the variant of errors here, not a possible stopped! That's what we have the next callback for.